### ISSUE TITLE
with ansi=false output is not sent via fast output (int 0x29)

### ISSUE THREAD

**Issue by stu** [joncampbell123/dosbox-x#2306]
When ansi.sys is disabled via config, but a tsr driver is loaded, any text output by the system is not sent via fast output int 0x29 it would seem.

**To Reproduce**
in config set ansi.sys=false, load an ansi driver tsr, then run and C application that does a printf that has ansi escape codes in its strings... you see the strings printed to screen. OR set your dos prompt to have ansi codes in it

set prompt=$e[0;37m $e[0;32m[ $e[0;36m$P$e[0;32m ]$e[0;37m$_$e[1;32m$s$g$s$e[0;37m

**Expected behavior**
with an ansi tsr loaded, it hooks fast char output int 0x29 and will support ansi codes.

**Environment (please complete the following information):**
 - Linux, ubuntu 20something
 - DOSBox-X release last commit 556c084d42516e2f81df805e06ece24f2ecaed9e

**Comment by stu** [comment]
So I did a test patch against CON for output, that works with 4dos, and apps that pump ansi out from eg printf in c. but in the dbx shell, it does not send its prompt out via CON it appears, so even with an ansi driver tsr installed, the prompt is messed up. 

So this issue would appear to go deeper than I expected.

```
diff --git a/src/dos/dev_con.h b/src/dos/dev_con.h
index 36f73e15b..98e0a3ee5 100644
--- a/src/dos/dev_con.h
+++ b/src/dos/dev_con.h
@@ -1267,7 +1267,15 @@ void device_CON::Output(uint8_t chr) {
 			}
 		}
 		Real_INT10_TeletypeOutputAttr(chr,ansi.attr,true);
-	} else Real_INT10_TeletypeOutput(chr,DefaultANSIAttr());
+	}
+	else
+	{
+		uint16_t oldax;
+		oldax=reg_ax;
+		reg_ax=chr;
+		CALLBACK_RunRealInt(0x29);
+		reg_ax=oldax;
+	}
 }
 
 bool device_CON::ANSI_SYS_installed() {
```

**Comment by Wengier** [comment]
Thanks for reporting this issue, and the test patch. The reason that the patch did not work for DOSBox-X's internal shell is that the case should be put before the dos.internal_output check, not after, or it will only affect the non-internal outputs. The issue is hopefully fixed as part of #2311.
